--- lizardfs-3.12.0/src/common/io_limits_config_loader.h.orig	2017-12-20 10:59:37.000000000 +0100
+++ lizardfs-3.12.0/src/common/io_limits_config_loader.h	2025-04-16 18:57:15.649316658 +0200
@@ -19,6 +19,7 @@
 #pragma once
 #include "common/platform.h"
 
+#include <cstdint>
 #include <istream>
 #include <map>
 #include <string>
--- lizardfs-3.12.0/src/mount/polonaise/setup.h.orig	2017-12-20 10:59:37.000000000 +0100
+++ lizardfs-3.12.0/src/mount/polonaise/setup.h	2025-04-16 19:01:20.787581683 +0200
@@ -19,6 +19,7 @@
 #pragma once
 #include "common/platform.h"
 
+#include <cstdint>
 #include <string>
 
 #include "protocol/MFSCommunication.h"
